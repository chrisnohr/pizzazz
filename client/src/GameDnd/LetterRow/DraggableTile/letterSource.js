export const dragCollect = (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
});

const letterSource = {
  beginDrag({ letter, index }) {
    return {
      sourceLetter: letter,
    };
  },
  endDrag(props, monitor) {
    if (monitor.didDrop()) {
      const target = monitor.getDropResult();
      const { endDragAction } = props;
      endDragAction(props, target);
    }
  },
};

export default letterSource;
