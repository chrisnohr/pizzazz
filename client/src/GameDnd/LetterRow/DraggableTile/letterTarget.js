export const dropCollect = connect => ({
  connectDropTarget: connect.dropTarget(),
});

const letterTarget = {
  drop({
    string,
    index,
    parent,
  }) {
    return {
      targetString: string,
      targetIndex: index,
      targetParent: parent,
    };
  },
};

export default letterTarget;
