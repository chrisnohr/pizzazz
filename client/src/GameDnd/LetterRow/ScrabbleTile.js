import React, { PureComponent } from 'react';
import { LetterStyled, NumberSub } from './ScrabbleTileStyled';
import { letterValues } from '../../Constants';

class ScrabbleTile extends PureComponent {
  render() {
    const { letter, isValid } = this.props;

    return (
      <LetterStyled isValid={isValid} letter={letter}>
        { letter.toUpperCase() }
        <NumberSub>
          { letterValues[letter] }
        </NumberSub>
      </LetterStyled>
    );
  }
}

export default ScrabbleTile;
