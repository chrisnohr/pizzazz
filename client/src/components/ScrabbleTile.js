// @flow
import React, { PureComponent } from 'react';
import type { Node } from 'react';
import { LetterStyled, NumberSub } from './styled/ScrabbleTileStyled';
import { letterValues } from '../Constants';

type TileProps = {
  isValid: boolean,
  letter: string,
};

class ScrabbleTile extends PureComponent<TileProps> {
  render(): Node {
    const { letter, isValid } = this.props;

    return (
      <LetterStyled isValid={isValid} letter={letter}>
        { letter.toUpperCase() }
        <NumberSub>
          { letterValues[letter] }
        </NumberSub>
      </LetterStyled>
    );
  }
}

export default ScrabbleTile;
