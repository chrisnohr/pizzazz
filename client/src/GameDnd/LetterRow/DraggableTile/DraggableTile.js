/* eslint no-shadow: off */
import React, { Component } from 'react';
import { DragSource, DropTarget } from 'react-dnd';
import { getEmptyImage } from 'react-dnd-html5-backend';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import ScrabbleTile from '../ScrabbleTile';
import { endDragAction } from '../../gameDndState';
import letterSource, { dragCollect } from './letterSource';
import letterTarget, { dropCollect } from './letterTarget';


class DraggableTile extends Component {
  componentDidMount() {
    const { connectDragPreview } = this.props;
    connectDragPreview(
      getEmptyImage(),
      { captureDraggingState: true },
    );
  }

  render() {
    const {
      letter,
      index,
      connectDragSource,
      connectDropTarget,
      isValid,
      isDragging,
    } = this.props;

    return connectDragSource(
      connectDropTarget(
        <div style={{ width: '13.68%' }}>
          <ScrabbleTile
            letter={isDragging ? '0' : letter}
            index={index}
            isValid={isValid}
          />
        </div>,
      ),
    );
  }
}

DraggableTile.propTypes = {
  letter: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  isValid: PropTypes.bool.isRequired,
  isDragging: PropTypes.bool.isRequired,
  endDragAction: PropTypes.func.isRequired, // eslint-disable-line
  connectDragSource: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  connectDragPreview: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  endDragAction: (props, target) => dispatch(endDragAction(props, target)),
});

export default compose(
  connect(null, mapDispatchToProps),
  DragSource('letter', letterSource, dragCollect),
  DropTarget('letter', letterTarget, dropCollect),
)(DraggableTile);

export const UnconnectedDraggableTile = compose(
  DragSource('letter', letterSource, dragCollect),
  DropTarget('letter', letterTarget, dropCollect),
)(DraggableTile);
