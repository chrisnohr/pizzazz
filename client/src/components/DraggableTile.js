/* eslint no-shadow: off */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { DragSource, DropTarget } from 'react-dnd';
import { getEmptyImage } from 'react-dnd-html5-backend';
import { compose } from 'redux';
import ScrabbleTile from './ScrabbleTile';
import { endDragAction } from '../actions/actionCreators';


class DraggableTile extends Component {
  componentDidMount() {
    const { connectDragPreview } = this.props;
    if (connectDragPreview) {
      connectDragPreview(getEmptyImage(), {
        captureDraggingState: true,
      });
    }
  }

  render() {
    const {
      letter,
      index,
      connectDragSource,
      connectDropTarget,
      isValid,
      isDragging,
    } = this.props;

    return connectDragSource(
      connectDropTarget(
        <div style={{ width: '13.68%' }}>
          <ScrabbleTile
            letter={isDragging ? '0' : letter}
            index={index}
            isValid={isValid}
          />
        </div>,
      ),
    );
  }
}

const letterSource = {
  beginDrag(props) {
    return {
      sourceLetter: props.letter,
      sourceIndex: props.index,
    };
  },
  endDrag(props, monitor) {
    if (monitor.didDrop()) {
      const target = monitor.getDropResult();
      const { endDragAction } = props;
      endDragAction(props, target);
    }
  },
};
const letterTarget = {
  drop(props) {
    const {
      letter, string, index, parent,
    } = props;
    return {
      targetLetter: letter,
      targetString: string,
      targetIndex: index,
      targetParent: parent,
    };
  },
};

const dragCollect = (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
});
const dropCollect = connect => ({
  connectDropTarget: connect.dropTarget(),
});

DraggableTile.propTypes = {
  letter: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  isValid: PropTypes.bool.isRequired,
  isDragging: PropTypes.bool.isRequired,
  endDragAction: PropTypes.func.isRequired, // eslint-disable-line
  connectDragSource: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  connectDragPreview: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  endDragAction: (props, target) => dispatch(endDragAction(props, target)),
});

export default compose(
  connect(null, mapDispatchToProps),
  DragSource('letter', letterSource, dragCollect),
  DropTarget('letter', letterTarget, dropCollect),
)(DraggableTile);
