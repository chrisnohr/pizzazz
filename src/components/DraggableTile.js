import React, { Component } from 'react'
import { connect } from 'react-redux'
import { DragSource, DropTarget } from 'react-dnd'
import { compose } from 'redux'
import { updateValidWord } from '../scrabbleLogic/gameLogic'
import { findValidWord, findValidWordWildcard } from '../scrabbleLogic/findWords'
import { wordScoreDict, wordScoreString, letterValues } from '../Constants'
import ScrabbleTile from './ScrabbleTile'

class DraggableTile extends Component {
  render() {
    const { letter, index, connectDragSource, connectDropTarget, showValid } = this.props;
    return connectDragSource(
              connectDropTarget(
                <div style={{ width: 13 + "%", }}>
                  <ScrabbleTile
                    letter={ letter }
                    index={ index }
                    letterValues= { letterValues }
                    className={showValid ? "is-valid " + "d-inline" : "" + "d-inline"}
                  />
                </div>
              )
            )
  }
}

const letterSource = {
  beginDrag(props) {
    return {
      sourceLetter: props.letter,
      sourceIndex: props.index,
    }
  },
  endDrag(props, monitor) {
    if (monitor.didDrop()) {
      const target = monitor.getDropResult();
      const { letter, index, parent, replaceLetterAction } = props;
      const { targetLetter, targetIndex, targetParent, showValidAction } = target;
      props.dispatch(replaceLetterAction(letter, targetParent, targetIndex));
      props.dispatch(replaceLetterAction(targetLetter, parent, index));



      if (targetParent === 'validWord' || parent === 'validWord') {
        const word = 0;
        const validWordString = updateValidWord(props, target);

        if (validWordString.indexOf('8') !== -1) {
          var validWords = findValidWordWildcard(validWordString, wordScoreString);
          if (validWords){
            var longestValidWord = validWords[word];
            // var longestValidWordScore = validWords[score];
          }
        }
        else {
          longestValidWord = findValidWord(validWordString, wordScoreDict)[word];
          // longestValidWordScore = findValidWord(validWordString, wordScoreDict)[score];
        }
        if (longestValidWord) {
          props.dispatch(showValidAction(longestValidWord.length))
        }
        else {
          props.dispatch(showValidAction(0));
        }
      }
    }
  }
}

const letterTarget = {
  drop(props) {
    const { letter, string, index, parent, showValidAction } = props;
    return {
      targetLetter: letter,
      targetString: string,
      targetIndex: index,
      targetParent: parent,
      showValidAction: showValidAction
    }
  }
}

function dragCollect(connect) {
  return {
    connectDragSource: connect.dragSource()
  }
}

function dropCollect(connect) {
  return {
    connectDropTarget: connect.dropTarget()
  }
}

export default compose(
  connect(),
  DragSource('letter', letterSource, dragCollect),
  DropTarget('letter', letterTarget, dropCollect),
)(DraggableTile);
