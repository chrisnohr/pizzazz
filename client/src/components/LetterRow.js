import React from 'react';
import PropTypes from 'prop-types';
import ScrabbleTile from './ScrabbleTile';
import DraggableTile from './DraggableTile';
import { LetterContainer } from './styled/ScrabbleTileStyled';

/* eslint react/no-array-index-key: off */
const LetterRow = ({
  isDraggable, letters, parent, isValidIndex,
}) => {
  if (isDraggable) {
    return letters.split('')
      .map((letter, i) => (
        <DraggableTile
          letter={letter}
          string={letters}
          isValid={isValidIndex > i}
          index={i}
          key={i}
          parent={parent}
        />
      ));
  }
  return letters.split('')
    .map((letter, i) => (
      <LetterContainer key={i}>
        <ScrabbleTile
          letter={letter}
          index={i}
          isValid={false}
        />
      </LetterContainer>
    ));
};
//     ? )
//     : letters.split('')
//       .map((letter, i) => (
//         <LetterContainer key={i}>
//           <ScrabbleTile
//             letter={letter}
//             index={i}
//             isValid={false}
//           />
//         </LetterContainer>
//       ))
// );

LetterRow.propTypes = {
  isDraggable: PropTypes.bool.isRequired,
  letters: PropTypes.string.isRequired,
  parent: PropTypes.string.isRequired,
  isValidIndex: PropTypes.number,
};

LetterRow.defaultProps = {
  isValidIndex: 0,
};

export default LetterRow;
