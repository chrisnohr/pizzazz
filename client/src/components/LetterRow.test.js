import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import LetterRow from './LetterRow';
import DraggableTile from './DraggableTile';
import ScrabbleTile from './ScrabbleTile';

describe('<LetterRow />', () => {
  it('renders ScrabbleTiles if isDraggable props is false', () => {
    const wrapper = shallow(
      (<LetterRow
        isDraggable={false}
        letters="pizzazz"
        parent="randomLetters"
        isValidIndex={7}
      />),
    );
    expect(wrapper.find(ScrabbleTile)).toHaveLength(7);
  });
  it('renders DraggableTiles if isDraggable props is true', () => {
    const wrapper = shallow(
      (<LetterRow
        isDraggable
        letters="pizzazz"
        parent="randomLetters"
        isValidIndex={7}
      />),
    );
    expect(wrapper.find(DraggableTile)).toHaveLength(7);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(
      (<LetterRow
        isDraggable
        letters="pizzazz"
        parent="randomLetters"
        isValidIndex={7}
      />),
    );
    expect(wrapper).toMatchSnapshot();
  });
});
