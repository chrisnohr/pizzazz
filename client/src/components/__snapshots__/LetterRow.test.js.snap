// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetterRow /> matches the snapshot 1`] = `
Array [
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={0}
    isValid={true}
    key="0"
    letter="p"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={1}
    isValid={true}
    key="1"
    letter="i"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={2}
    isValid={true}
    key="2"
    letter="z"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={3}
    isValid={true}
    key="3"
    letter="z"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={4}
    isValid={true}
    key="4"
    letter="a"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={5}
    isValid={true}
    key="5"
    letter="z"
    parent="randomLetters"
    string="pizzazz"
  />,
  <Connect(DragSource(DropTarget(DraggableTile)))
    index={6}
    isValid={true}
    key="6"
    letter="z"
    parent="randomLetters"
    string="pizzazz"
  />,
]
`;
